src/compiler/factory/nodeConverters.ts

```diff
@@ -57,7 +57,7 @@ namespace ts {
                     )
                     : expression;
             }
-            return cast(element, isExpression);
+            return cast<Expression>(element, isExpression); // TODO(jakebailey): GH#49924 ???

```

Comment:
Lots of weird type arguments like these are need; see #49924.

I can apply the change suggested there, but it negatively affects other parts of the code.